### PR TITLE
docs/multiheap: Add supported APIs description

### PR DESCRIPTION
**Pull request by jeongchanKim**
Xalloc_at APIs can be used to allocate memory for a specific heap.

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [a4d0df0bad5be040e2d741db5321deb5b75454a0] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [a4d0df0bad5be040e2d741db5321deb5b75454a0] - Code Rule Check OK.

File: docs/HowToUseMultiHeap.md [hunk 1]
```diff
@@ -57,3 +58,12 @@ CONFIG_RAM_REGIONx_SIZE="1024,512,2048,512,256,4096"
 CONFIG_RAM_REGIONx_HEAP_INDEX="0,1,1,2,2,3"
 ```
 
+## 4. Supported APIs
+Following APIs can be used to allocate memory for a specific heap.
```
Comment (sunghan-chang) [inline docs/HowToUseMultiHeap.md]: 1. Could you leave the header file name?
2. Could you leave descriptions for original APIs? When I use malloc on multi heap, how does it work against malloc_at?

**Comment by seinfra** [comment]
Target : [01677e7520c69e4b488c31f096cb75645eb1d9d8] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [01677e7520c69e4b488c31f096cb75645eb1d9d8] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [964f4c2a755af327df62ea388f3e50352a21e313] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [964f4c2a755af327df62ea388f3e50352a21e313] - Code Rule Check (C++) OK.

File: docs/HowToUseMultiHeap.md [hunk 2]
```diff
@@ -8,6 +8,7 @@ To use multi-heap, please find three steps as shown below:
 - [Set number of Regions](#1-set-number-of-regions)
 - [Set number of Heaps](#2-set-number-of-heaps)
 - [Set Region start address, size and heap index](#3-set-region-start-address-size-and-heap-index)
+- [Supported APIs](#4-supported-apis)
```
Comment (sunghan-chang) [inline docs/HowToUseMultiHeap.md]: We need to split this step. This step is for usage and above are for enabling.

File: docs/HowToUseMultiHeap.md [hunk 3]
```diff
@@ -57,3 +58,16 @@ CONFIG_RAM_REGIONx_SIZE="1024,512,2048,512,256,4096"
 CONFIG_RAM_REGIONx_HEAP_INDEX="0,1,1,2,2,3"
 ```
 
+## 4. Supported APIs
```
Comment (sunghan-chang) [inline docs/HowToUseMultiHeap.md]: Need to remove numbering.

File: docs/HowToUseMultiHeap.md [hunk 4]
```diff
@@ -57,3 +58,16 @@ CONFIG_RAM_REGIONx_SIZE="1024,512,2048,512,256,4096"
...
+Following APIs support to allocate memory for a specific heap.  
+A header file [mm.h](../os/include/tinyara/mm/mm.h) provides the API descriptions.
```
Comment (sunghan-chang) [inline docs/HowToUseMultiHeap.md]: A header file [mm.h] provides following APIs which supports to allocate memory for a specific heap region as shown below:

**Comment by seinfra** [comment]
Target : [06d33686d198487e045a769a27bc86dbf572c25c] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [06d33686d198487e045a769a27bc86dbf572c25c] - Code Rule Check (C++) OK.

File: docs/HowToUseMultiHeap.md [hunk 5]
```diff
@@ -4,21 +4,24 @@ There are two cases to use multi-heap.
 1. continuous physical RAM but want to split regions
 2. separated physical RAMs but want to use for heap  
 
-To use multi-heap, please find three steps as shown below:
+To enable multi-heap, please find three steps as shown below:
 - [Set number of Regions](#1-set-number-of-regions)
 - [Set number of Heaps](#2-set-number-of-heaps)
 - [Set Region start address, size and heap index](#3-set-region-start-address-size-and-heap-index)
 
+Following APIs support to allocate memory for a specific heap.
```
Comment (sunghan-chang) [inline docs/HowToUseMultiHeap.md]: To use multi-heap, please find a link as shown below:

File: docs/HowToUseMultiHeap.md [hunk 6]
```diff
@@ -4,21 +4,24 @@ There are two cases to use multi-heap.
...
 - [Set number of Heaps](#2-set-number-of-heaps)
 - [Set Region start address, size and heap index](#3-set-region-start-address-size-and-heap-index)
 
+Following APIs support to allocate memory for a specific heap.  
+[Supported APIs](#supported-apis)
+
 ## 1. Set number of Regions
 
-Set **CONFIG_MM_REGIONS** value which shows how many regions are.
+Set **CONFIG_MM_REGIONS** value in menuconfig which shows how many regions are.
```
Comment (sunghan-chang) [inline docs/HowToUseMultiHeap.md]: Is this right location to add `in menuconfig`?

**Comment by seinfra** [comment]
Target : [757dbc76930c8687d18aa0c7acfc788c397a61f0] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [757dbc76930c8687d18aa0c7acfc788c397a61f0] - Code Rule Check OK.

File: docs/HowToUseMultiHeap.md [hunk 7]
```diff
@@ -28,14 +31,14 @@ Memory Management -> Number of heaps -> change a number over 2
...
-Set start addresses, **CONFIG_RAM_REGIONx_START** with hexa values,  set sizes, **CONFIG_RAM_REGIONx_SIZE** with decimal values(in bytes) of new heap, and set indexes, **CONFIG_RAM_REGIONx_HEAP_INDEX** with decimal values of new heap.  
+Set start addresses, **CONFIG_RAM_REGIONx_START** with hexa values,  set sizes, **CONFIG_RAM_REGIONx_SIZE** with decimal values(in bytes) of new heap, and set indexes, **CONFIG_RAM_REGIONx_HEAP_INDEX** with decimal values of new heap in menuconfig.  
 **CONFIG_RAM_REGIONx_HEAP_INDEX** can be start from 0.
 ```
 Hardware Configuration -> Chip Selection -> List of start address for RAM region -> set values
 Hardware Configuration -> Chip Selection -> List of size for RAM region
 Hardware Configuration -> Chip Selection -> List of heap index for RAM region
 ```
-Each region is separated by ```','``` as shown below example:
+Each region is separated by ```','``` and all config should be in ```" "``` as shown below example:
```
Comment (sunghan-chang) [inline docs/HowToUseMultiHeap.md]: Please check an example at below

**Comment by seinfra** [comment]
Target : [8a49d41b2b61814e8cd1b906139b52d9b65df8cc] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [8a49d41b2b61814e8cd1b906139b52d9b65df8cc] - Code Rule Check OK.

**Pull request merged**